commitizen/commands/commit.py

```diff
@@ -63,10 +64,18 @@ def prompt_commit_questions(self) -> str:
 
     def __call__(self):
         dry_run: bool = self.arguments.get("dry_run")
+        write_message_to_file = self.arguments.get("write_message_to_file")
 
         if git.is_staging_clean() and not dry_run:
             raise NothingToCommitError("No files added to staging!")
 
+        if write_message_to_file is not None:
```

Comment:
Can we merge this 2 lines?

```python
if write_message_to_file is not None and not isinstance(write_message_to_file, str)
```

I'm wondering if we should also check if it's a file 🤔 I guess the `smart_open` will raise an error

Reply:
`smart_open` in line 89 does raise an error if the path points to an existing directory but, considering that the user had to go through answering the questions at that point, it makes sense to already raise an error here. does the cli library support `pathlib.Path` as a type?

if not, i'd add a check like this:

```python
if write_message_to_file is not None:
    if not isinstance(write_message_to_file, str):
        raise NotAllowed(...)
    
    if os.path.isdir(write_message_to_file):
        raise NotAllowed(...)
```

Reply:
https://github.com/commitizen-tools/commitizen/pull/731#discussion_r1181007836